### PR TITLE
Civ 6582 defendant cant afford

### PR DESCRIPTION
**Pull request by GorilaNaranja**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-6582

### Change description ###
Claimant Intent – The defendant can't afford your plan

### URL ###
http://localhost:3001/case/1645882162449409/claimant-response/court-offered-instalments

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/common/utils/dateUtils.ts [hunk 1]
```diff
@@ -52,3 +52,8 @@ export const getDOBforAgeFromCurrentTime = (age: number): Date => {
   referenceDate.setFullYear(referenceDate.getFullYear() - age);
   return referenceDate;
 };
+
```
Comment (GorilaNaranja) [inline src/main/common/utils/dateUtils.ts:55]: This file comes from:
https://tools.hmcts.net/jira/browse/CIV-6375

File: src/main/common/utils/repaymentUtils.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import {Claim} from 'common/models/claim';
```
Comment (GorilaNaranja) [inline src/main/common/utils/repaymentUtils.ts:1]: This file comes from https://tools.hmcts.net/jira/browse/CIV-6375 **EXCEPT last function convertFrequencyToText()** that is new and should be reviwed

File: src/test/unit/common/utils/repaymentUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
+import {HowMuchDoYouOwe} from 'common/form/models/admission/partialAdmission/howMuchDoYouOwe';
```
Comment (GorilaNaranja) [inline src/test/unit/common/utils/repaymentUtils.test.ts:1]: This file comes from:
https://tools.hmcts.net/jira/browse/CIV-6375

**Except convertFrequencyToText test that is new**

File: sonar-project.properties [hunk 4]
```diff
@@ -10,7 +10,7 @@ sonar.exclusions=src/main/app.ts,src/main/development.ts,src/main/index.js,src/m
 src/main/assets/js/postcode-lookup.js,src/main/assets/js/calculate-amount.js,src/main/assets/js/language-toggle.js,src/main/assets/js/calculate-length-repayment.js,src/main/assets/scss/task-list.scss,\
 src/main/common/utils/taskList/taskListBuilder.ts,src/main/common/utils/taskList/tasks/confirmYourDetails.ts,\
 src/main/modules/taskListService.ts,\
-src/main/routes/features/claimantResponse/chooseHowToProceedcController.ts,src/main/routes/features/response/taskListController.ts,src/main/assets/js/conditionally-hide-add-button.js,\
+src/main/routes/features/claimantResponse/courtProposedPlanController.ts,src/main/routes/features/claimantResponse/chooseHowToProceedcController.ts,src/main/routes/features/response/taskListController.ts,src/main/assets/js/conditionally-hide-add-button.js,\
```
Comment (TatsyMaslova) [inline sonar-project.properties:13]: sorry Felipe how come we are not including courtProposedplanController.ts into sonar?

File: src/main/common/utils/repaymentUtils.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import {addDaysToDate, addMonths} from './dateUtils';
+import {TransactionSchedule} from 'common/form/models/statementOfMeans/expensesAndIncome/transactionSchedule';
+import {t} from 'i18next';
+
+const WEEKDAYS = 7;
+let paymentAmount = 0;
+let repaymentFrequency = '';
+let firstRepaymentDate = new Date(Date.now());
+
+export const isRepaymentPlanFullOrPartAdmit = (claim: Claim) => {
```
Comment (TatsyMaslova) [inline src/main/common/utils/repaymentUtils.ts:11]: I am not sure the name of the function. 'is' prefix usually indicates that it should return a boolean. But instead it is setting global variables

File: src/main/common/utils/repaymentUtils.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    firstRepaymentDate = new Date(repaymentPlan?.firstRepaymentDate);
+  } else if (claim.isPartialAdmission()) {
+    const repaymentPlan = claim.partialAdmission.paymentIntention.repaymentPlan;
+    paymentAmount = repaymentPlan?.paymentAmount;
+    repaymentFrequency = repaymentPlan?.repaymentFrequency;
+    firstRepaymentDate = new Date(repaymentPlan?.firstRepaymentDate);
+  }
+};
+
+export const getNumberOfInstalments = (claim: Claim) => {
```
Comment (TatsyMaslova) [inline src/main/common/utils/repaymentUtils.ts:25]: missing return type

File: src/main/common/utils/repaymentUtils.ts (hunk 5 above)
Comment (GorilaNaranja) [reply src/main/common/utils/repaymentUtils.ts:11]: This code is not from my PR, it comes from https://tools.hmcts.net/jira/browse/CIV-6375. 
@RosDiaf if you are agree we can change it in both PRs.

File: src/main/common/utils/repaymentUtils.ts (hunk 6 above)
Comment (GorilaNaranja) [reply src/main/common/utils/repaymentUtils.ts:25]: Same, this code is not from this PR but I will add return type to the function that I created (convertFrequencyToText).
@RosDiaf can you fix that for your code?

**Review by laperla11 — APPROVED** [review]

File: sonar-project.properties (hunk 4 above)
Comment (TatsyMaslova) [reply sonar-project.properties:13]: okie

File: src/main/modules/i18n/locales/en.json [hunk 7]
```diff
@@ -684,6 +689,23 @@
       "TITLE": "What annual rate of interest do you want to claim?",
       "WHY_YOU_ARE_ENTITLED": "Why you're entitled to this rate"
     },
+    "CLAIMANT_RESPONSE": {
```
Comment (portizdev) [inline src/main/modules/i18n/locales/en.json:692]: The lines number in the file en and cy do not match.

Comment (GorilaNaranja) [reply src/main/modules/i18n/locales/en.json:692]: We had some inconsistences in translations files but I fixed it.

File: src/main/modules/i18n/locales/cy.json [hunk 8]
```diff
@@ -313,6 +318,11 @@
       "ENGLISH": "English",
       "WELSH_AND_ENGLISH": "Welsh and English"
     },
+    "BREATHING_SPACE_DEBT_RESPITE_START": {
```
Comment (GorilaNaranja) [inline src/main/modules/i18n/locales/cy.json:321]: Not related to this ticket but I added missing translations to have en.json and cy.json equal.

File: src/main/modules/i18n/locales/cy.json [hunk 9]
```diff
@@ -2146,6 +2174,7 @@
     "SELECT_YES_IF_DOCUMENTS": "Dewiswch ‘oes’ os oes gan yr hawlydd ddogfennau yr ydych eisiau i’r llys eu hystyried",
     "SELECT_YES_IF_SUPPORT": "Select yes if someone needs support to attend a hearing",
     "SIGNER_NAME_REQUIRED": "Enter the name of the person signing the statement",
+    "SIGNER_NAME_TOO_LONG": "You've entered too many characters",
```
Comment (GorilaNaranja) [inline src/main/modules/i18n/locales/cy.json:2177]: Not related to this ticket but I added missing translations to have en.json and cy.json equal.

File: src/main/modules/i18n/locales/en.json [hunk 10]
```diff
@@ -711,6 +733,11 @@
       "PAGE_TITLE": "When did you settle with the defendant?",
       "TITLE": "When did you settle with the defendant?"
     },
+    "CLAIMANT_RESPONSE_DEBT_WHEN_START": {
```
Comment (GorilaNaranja) [inline src/main/modules/i18n/locales/en.json:736]: Not related to this ticket but I added missing translations to have en.json and cy.json equal.

**Review by portizdev — APPROVED** [review]

File: src/main/common/utils/repaymentUtils.ts (hunk 5 above)
Comment (GorilaNaranja) [reply src/main/common/utils/repaymentUtils.ts]: File refactored @TatsyMaslova

File: src/main/common/utils/repaymentUtils.ts (hunk 6 above)
Comment (RosDiaf) [reply src/main/common/utils/repaymentUtils.ts:8]: method refactored

File: src/main/common/utils/repaymentUtils.ts (hunk 5 above)
Comment (RosDiaf) [reply src/main/common/utils/repaymentUtils.ts]: method refactored

File: src/main/common/utils/repaymentUtils.ts (hunk 6 above)
Comment (GorilaNaranja) [reply src/main/common/utils/repaymentUtils.ts]: Also, I added return types

**Review by TatsyMaslova — APPROVED** [review]

**Pull request merged**